prog.go

```diff
@@ -51,9 +51,12 @@ type ProgramSpec struct {
 	// Type determines at which hook in the kernel a program will run.
 	Type       ProgramType
 	AttachType AttachType
-	// Name of a kernel data structure to attach to. It's interpretation
-	// depends on Type and AttachType.
-	AttachTo     string
+	// Name of a kernel data structure or function to attach to. Its
+	// interpretation depends on Type and AttachType.
+	AttachTo string
+	// The file descriptor of the target. Its interpretation depends on Type
+	// and AttachType. Must be manually specified.
+	AttachTarget int
```

Comment:
Libbpf takes it as an int. We could specify it as a Program, but I believe maps can (or soon will be able to be) valid attach targets.

Reply:
The reason I believe maps as valid attach targets is likely coming is from a conversation on the kernel bpf mailing list where having functions attached to maps and trigger on element updates, creation, removals, etc. was mentioned. I'm trying to find the conversation that mentioned that, but I can't remember the proper keywords to search, so I haven't yet.

Reply:
Found the conversation: https://lore.kernel.org/bpf/CAADnVQLWj-=B2TfJp7HEsiUY3rqmd6-YMDAGdyL6RgZ=_b2CXg@mail.gmail.com/

Reply:
It's possible this will not require the AttachTarget functionality here, but also possible it will. Since libbpf takes an int and not a `struct bpf_progragm`, I take that as an indication that future flexibility is desired.

Reply:
Was thinking about this (before reading this comment of yours) and came to basically the same conclusion. Plus, if attaching to something that isn't a program ever becomes necessary, we can potentially create an interface that all types that can be freplaced implement, allowing us to add support in a non-breaking way. Or make per-object-type functions. Or we can just make a breaking change and release a new major version. Lots of options.

Reply:
Addressed in 80be84a